state/factory/statetx.go

```diff
@@ -232,6 +232,23 @@ func (stx *stateTX) State(hash hash.Hash160, s interface{}) error {
 	return state.Deserialize(s, mstate)
 }
 
+// StateAtHeight pulls a state from DB
+func (stx *stateTX) StateAtHeight(height uint64, addr hash.Hash160, s interface{}) error {
+	ns := append([]byte(AccountKVNameSpace), addr[:]...)
+	ri, err := db.NewRangeIndex(stx.dao, ns, db.NotExist)
+	if err != nil {
+		return err
+	}
+	data, err := ri.Get(height)
+	if err != nil {
+		if errors.Cause(err) == db.ErrNotExist {
+			return errors.Wrapf(state.ErrStateNotExist, "state of %x doesn't exist", addr)
+		}
+		return errors.Wrapf(err, "error when getting the state of %x", addr)
+	}
+	return state.Deserialize(s, data)
+}
```

Comment:
as commented previously, we don't need to implement this one. Instead, we should throw an error "not supported". StateTX and StateDB are not designed for archive mode.